#!/usr/bin/env python3
PACKAGE = "qr_navigation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("forward_max_vel",    double_t,    0, "max linear vel", 0.7,  0.05, 0.8)
gen.add("reverse_max_vel",    double_t,    0, "reverse_max_vel", 0.7,  0.05, 0.8
)
gen.add("homing_distance",    double_t,    0, "distance to enable extractor homing ", 0.8,   0.05, 2)

gen.add("acc_constant", double_t, 0, "acceleration parameter ", 0.1, 0.1, 1)
gen.add("dec_constant",   double_t,    0, "deacceleration parameters", 0.1, 0.1,1)
gen.add("inertia_constant", double_t,0,"inertia of the robot to be consider", 0.5, 0.1,1)
gen.add("x_position_tolerance", double_t,0,"position accuracy to be achived near goal ",0.04, 0.005,0.1)
gen.add("x_angular_tolerance", double_t,0,"angular accuracy to be achived near goal ",0.04, 0.005,0.2)


gen.add("acceleration", double_t , 0, "acceleration", 0.5, 0.1, 1.5)

gen.add("min_lookahead_dist", double_t , 0, "min_lookahead_dist", 0.5, 0.1, 1.5)
gen.add("max_lookahead", double_t , 0, "max_lookahead", 1.0, 0.5, 1.5)
gen.add("wheel_base",	double_t, 0, "wheel_base",0.5,0.4,1.5)
gen.add("max_angular_vel", double_t,0,",max_angular_vel",0.2,0.5,1.5)

gen.add("kla", double_t, 0, "kla parameter", 1.1, 1, 2)


gen.add("charge_vel", double_t ,0 ,"charging linear velocity" ,0.1, 0.05, 1)
gen.add("dock_vel", double_t ,0 ,"docking linear velocity" ,0.1, 0.05, 1)


gen.add("inplace_max_rot_vel", double_t ,0 ,"inplace max rot velocity" ,0.2, 0,2) 
gen.add("control_gain", double_t ,0 ,"control gain" ,0.7 ,0 , 1)

exit(gen.generate(PACKAGE, "qr_navigation", "Parameter"))
